python/ray/_raylet.pyx

```diff
@@ -287,7 +287,13 @@ cdef void prepare_args(
         else:
             serialized_arg = worker.get_serialization_context().serialize(arg)
             size = serialized_arg.total_bytes
-            if <int64_t>size <= put_threshold:
+
+            # TODO(edoakes): any objects containing ObjectIDs are spilled to
+            # plasma here. This is inefficient for small objects, but inlined
+            # arguments aren't associated ObjectIDs right now so this is a
+            # simple fix for reference counting purposes.
```

Comment:
Hmm why is this necessary? I think we only need to do this for object IDs that were contained in another ObjectID (which we did [here](https://github.com/ray-project/ray/blob/master/src/ray/core_worker/core_worker.cc#L333)), right?

Reply:
Ah nvm, I understand now. Yeah, it is a bit unfortunate, but I guess later we can fix it by assigning inlined objects ObjectIDs as well. Actually shouldn't be too hard since we have the in-memory store.

Reply:
Yep, I think the best solution is to just `Put()` everything and put in in-memory store/plasma based on size in the core worker.